File: wordflip-server/src/main/java/com/wordflip/dto/group/CreateCustomGroupRequest.java
package com.wordflip.dto.group;

import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.Size;
import java.util.List;

/**
 * 使用当前计划未入组卡片创建手动分组。
 */
public record CreateCustomGroupRequest(
        @NotEmpty @Size(min = 1, max = 500) List<Long> cardIds,
        @Size(max = 64) String name
) {
}

File: wordflip-android/core-model/src/main/java/com/wordflip/core/model/group/CreateCustomGroupRequest.kt
package com.wordflip.core.model.group

/**
 * 创建自定义分组请求，对齐 openapi `CreateCustomGroupRequest`。
 * cardIds 须来自当前计划未入组卡片池。
 */
data class CreateCustomGroupRequest(
    val cardIds: List<Long>,
    val name: String? = null,
)
